File: Tool/PreUtil/src/output/python/hard/MedianOfTwoSortedArrays.py
#https://leetcode.com/problems/median-of-two-sorted-arrays/
#Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
#The overall run time complexity should be O(log (m+n)).


class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """       
        nums1.extend(nums2)
        nums1.sort()
        if len(nums1)%2==0:
            return (nums1[len(nums1)//2]+nums1[len(nums1)//2-1])/2
        else:
            return nums1[len(nums1)//2]

        
    

File: Tool/PreUtil/src/output/java/hard/MedianOfTwoSortedArrays.java
/**https://leetcode.com/problems/median-of-two-sorted-arrays/ */
//Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
//The overall run time complexity should be O(log (m+n)).


class MedianOfTwoSortedArrays {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] nums3 = new int[nums1.length + nums2.length];
        int i = 0, j = 0, k = 0;
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] < nums2[j]) {
                nums3[k] = nums1[i];
                i++;
            } else {
                nums3[k] = nums2[j];
                j++;
            }
            k++;
        }
        while (i < nums1.length) {
            nums3[k] = nums1[i];
            i++;
            k++;
        }
        while (j < nums2.length) {
            nums3[k] = nums2[j];
            j++;
            k++;
        }
        if (nums3.length % 2 == 0) {
            return (nums3[nums3.length / 2] + nums3[nums3.length / 2 - 1]) / 2.0;
        } else {
            return nums3[nums3.length / 2];
        }
        
    }
}
     
    